### PR TITLE
docs: explain using refs in a config file

### PR DESCRIPTION
**Pull request by tatomyr**
## What/Why/How?

Added docs for $refs in the config file: https://github.com/Redocly/redocly-cli/pull/1345.

Merge this alongside [the release](https://github.com/Redocly/redocly-cli/pull/1361).

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows (internal)
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -227,3 +227,23 @@ resolve:
 
 The first match takes priority when a URL matches multiple patterns.
 Therefore, only the header from the first match is used in the request.
+
+### Split up the configuration file
+
+As your config file grows, you may want to split it into multiple parts.
+That is possible by using references in a config similar to how they are used in OpenAPI descriptions:
```
Comment (HCloward) [inline docs/configuration/index.md:234]:
```suggestion
Splitting a config file is possible by using references in a config similar to how they are used in OpenAPI descriptions:
```

File: docs/configuration/index.md [hunk 2]
```diff
@@ -227,3 +227,23 @@ resolve:
...
+
+theme:
+  openapi:
+    $ref: ./openapi-theme.yaml
+  mockServer:
+    $ref: ./mockserver.yaml
+```
+
+{% admonition type="attention" %}
+When using the `push` command with `$ref`s in a config file, all referenced files should explicitly upload using the `--files` option
```
Comment (HCloward) [inline docs/configuration/index.md:248]:
```suggestion
When using the `push` command with a config file that includes `$ref`s , all referenced files are explicitly uploaded using the `--files` option.
```

**Review by HCloward — APPROVED** [review]
Made a couple of suggestions, otherwise looks good to me.

File: docs/configuration/index.md (hunk 2 above)
Comment (tatomyr) [reply docs/configuration/index.md:248]: @HCloward would the following work instead?

```suggestion
When using the `push` command with a config file that includes `$ref`s, all referenced files should be explicitly uploaded using the `--files` option.
```

Comment (HCloward) [reply docs/configuration/index.md:248]: yep, that works - why "should" though -  is there a possibility it won't work for some reason?

Comment (tatomyr) [reply docs/configuration/index.md:248]: Not really. It means that users should manually specify all the parts that need to be uploaded in the `--files` option. Do you think "are" fits here better?

Comment (tatomyr) [reply docs/configuration/index.md:248]: Ok, thanks for the clarification!

**Pull request merged**